src/templates/base.html

```diff
@@ -136,23 +142,91 @@
     </script>
       {% endcomment %}
       {% comment %}
-    Dynammically change where the search bar sends a GET request to based on
-    the selected value of the query type dropdown menu.
+      Dynammically change the autocomplete optins and where the search bar sends a GET
+      request to based on the selected value of the query type dropdown menu.
       {% endcomment %}
-      <script>
+      {# disable djlint for this section because the linter breaks the template tags #}
+      <!-- djlint:off --><script>   // "djlint:off" turns the linter off for this section of code
         document.addEventListener('DOMContentLoaded', function() {
           const search_bar = document.getElementById('search-bar');
           const query_type = document.getElementById('query-type');
 
-          query_type.addEventListener("change", function() {
+          // by default, query autocompletes game names
+          {% get_games as games %}
+          var game_names = [
+            {% for game in games %}
+              {
+                url: "{% url 'game-detail' pk=game.pk %}",
+                label: "{{game.name}}"
+              },
+            {% endfor %}
+          ];
+          $("#query").autocomplete({
+              source: function(request, response) {
+                /*
+                new regular expression object (RegExp) to search for pattern
+                request.term is the string entered into the search bar
+                "i" means that the matching is case insensitive
+                */
+                var matcher = new RegExp($.ui.autocomplete.escapeRegex(request.term), "i");
+                // results is all of the game names that match the term in the search bar
+                var results = $.grep(game_names, function(tag){
+                  return matcher.test(tag.label)
+                })
+                // only display the first 10 results
+                response(results.slice(0, 10));
```

Comment:
Minor unnecessary stylistic idea, but we could consider creating a constant that we set equal to 10 and have that constant here to improve readability and make it easier if we ever want to change the number of results we are displaying.

Reply:
yes I agree 100%, avoiding magic numbers and instead having this be stored as some `AUTOCOMPLETION_RESULTS_LIMIT` constant.